plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java

```diff
@@ -727,12 +729,185 @@ public void testTimestamp(ZoneId sessionZone)
+                .addRoundTrip("TIMESTAMP(6)", "TIMESTAMP '2020-09-27 12:34:56.123000'", createTimestampType(6), "TIMESTAMP '2020-09-27 12:34:56.123000'")
+                .addRoundTrip("TIMESTAMP(3)", "TIMESTAMP '2020-09-27 12:34:56.999'", createTimestampType(3), "TIMESTAMP '2020-09-27 12:34:56.999'")
+                .addRoundTrip("TIMESTAMP(7)", "TIMESTAMP '2020-09-27 12:34:56.1234567'", createTimestampType(7), "TIMESTAMP '2020-09-27 12:34:56.1234567'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123456789'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456789'")
+
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.000000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.1'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.100000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.9'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.900000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123000'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.999'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.999000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.1234567'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123456700'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123456789'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123456789'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.1'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.100000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.9'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.900000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123000'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.999'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.999000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.1234567'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456700'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123456789'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456789'")
+
```

Comment:
what happens when we use unsupported precision? like 10, 11, 12? should we test it?

Reply:
This concrete line is executed only for 
` .execute(getQueryRunner(), oracleCreateAndInsert("test_ts_oracle"));`

Because we create column directly in Oracle, it will be just not possible to create column with precision more then 9.

But it possible to create such case from trino and we have test for this:
`  .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.111222333444'", "TIMESTAMP '1970-01-01 00:00:00.111222333'")`

So trino will round 12 precision to 9, when it will be saved to Oracle